scss/_patterns_form-validation.scss

```diff
@@ -18,261 +18,110 @@
     margin-top: -$sp-unit;
   }
 
-  // Theming
-  @if ($theme-default-forms == 'dark') {
+  .is-caution {
     .p-form-validation__input {
-      .is-success & {
-        @include vf-form-validation-input-success-dark-theme;
-      }
+      background-color: $colors--theme--background-caution-default;
+      border-bottom-color: $colors--theme--border-caution;
 
-      .is-caution & {
-        @include vf-form-validation-input-caution-dark-theme;
+      &:hover {
+        background-color: $colors--theme--background-caution-hover;
       }
 
-      .is-error & {
-        @include vf-form-validation-input-error-dark-theme;
+      &:focus {
+        background-color: $colors--theme--background-caution-active;
       }
     }
 
     .p-form-validation__message {
-      .is-success & {
-        @include vf-form-validation-message-success-dark-theme;
-      }
-
-      .is-caution & {
-        @include vf-form-validation-message-caution-dark-theme;
+      @include vf-icon-warning($color: $color-caution);
+      color: $colors--theme--border-caution;
+
+      // FIXME: Temporary workaround for unsupported theme colors in icons
+      // stylelint-disable-next-line selector-max-type
+      body.is-dark &,
+      &.is-dark {
+        @include vf-icon-warning($color: #c48831, $color-symbol: $colors--dark-theme--background-default);
```

Comment:
If the icon color can be taked from variable it would be good to do it. I guess it's the tinted border (you may need to use the map-get to get the value out of the map.